### PR TITLE
Importing a pipeline by its name results in its name as the id

### LINKED ISSUE #265: Importing a pipeline by its name results in its name as the id
```
Terraform v0.12.24
+ provider.heroku v2.4.0
```

### Affected Resource(s)
`heroku_pipeline`

### Terraform Configuration Files
```hcl
resource "heroku_pipeline"  "some_pipeline" {
  name = "some_pipeline"
}
```

When I import the pipeline from heroku by its name it uses the pipeline `name` as the `ID` instead of the heroku provided UUID.

I replaced my pipeline name with "some_pipeline".

### Expected Behavior
`terraform import heroku_pipeline.main_pipeline some_pipeline` to set the `heroku_pipeline.main_pipeline.id` to eq the heroku supplied UUID.

### Actual Behavior
`terraform import heroku_pipeline.main_pipeline some_pipeline` results in this state:

```hcl
$ terraform state show heroku_pipeline.some_pipeline
resource "heroku_pipeline" "some_pipeline" {
    id   = "some_pipeline"
    name = "some_pipeline"

    owner {
        id   = "[REDACTED]"
        type = "team"
    }
}
```
### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Create a pipeline in heroku
2. Create a terraform file describing the previously created heroku pipeline
3. Run `terraform import heroku_pipeline.<resource_name> <your heroku pipeline *name*>`

### Important Factoids
When importing I used the heroku pipeline name instead of the ID. I'm guessing this was the cause.

Importing it by the UUID does result in the proper state.

### PR DESCRIPTION
**Pull request by davidji99**
Resolves #265 

Also bumped `terraform-plugin-sdk` to `v1.12.0`.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -34,11 +34,14 @@ github.com/bgentry/speakeasy v0.1.0 h1:ByYyxL9InA1OWqxJqqp2A5pYHUrCiAL6K3J+LKSsQ
 github.com/bgentry/speakeasy v0.1.0/go.mod h1:+zsyZBPWlz7T6j88CTgSN5bM796AkVf0kBD4zp0CCIs=
 github.com/cenkalti/backoff v2.1.1+incompatible h1:tKJnvO2kl0zmb/jA5UKAt4VoEVw1qxKWjE/Bpp46npY=
 github.com/cenkalti/backoff v2.1.1+incompatible/go.mod h1:90ReRw6GdpyfrHakVjL/QHaoyV4aDUVVkXQJJJ3NXXM=
+github.com/census-instrumentation/opencensus-proto v0.2.1/go.mod h1:f6KPmirojxKA12rnyqOA5BBL4O983OfeGPqjHWSTneU=
```
Comment (mars) [inline go.sum:37]: These are some odd dependencies to appear… all from the Terraform SDK upgrade?

File: heroku/resource_heroku_pipeline.go [hunk 2]
```diff
@@ -65,9 +65,11 @@ func resourceHerokuPipelineImport(d *schema.ResourceData, meta interface{}) ([]*
 		return nil, err
 	}
 
-	d.Set("name", p.Name)
+	d.SetId(p.ID)
+
+	readErr := resourceHerokuPipelineRead(d, meta)
```
[the remark this answers is not in the document]
Comment (davidji99) [reply heroku/resource_heroku_pipeline.go:70]: `resourceHerokuPipelineRead` doesn't set the resource id or call `d.SetId()`. During resource importation, `d.Id()` is whatever the user passed into the `terraform import` command so `d.Id()` could return the pipeline name or UUID. So I need to use first `GET` the pipeline, set the resource id to its UUID, then call `resourceHerokuPipelineRead` to set the other resource attributes.

[the remark this answers is not in the document]
Comment (davidji99) [reply heroku/resource_heroku_pipeline.go:70]: I could just extract out https://github.com/terraform-providers/terraform-provider-heroku/blob/master/heroku/resource_heroku_pipeline.go#L170-L175 into a separate function that the import function also calls in order to set the resource attributes.

File: go.sum (hunk 1 above)
Comment (davidji99) [reply go.sum:37]: Yeah. This package is also present in https://github.com/terraform-providers/terraform-provider-aws/blob/master/go.sum#L64

Comment (mars) [reply go.sum:37]: Thanks for checking 🙏

File: heroku/resource_heroku_pipeline.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (mars) [reply heroku/resource_heroku_pipeline.go]: Looks good!

**Pull request merged**